bundles/org.openhab.voice.whisperstt/src/main/java/org/openhab/voice/whisperstt/internal/WhisperSTTService.java

```diff
@@ -0,0 +1,656 @@
+                        // process result
+                        if (result != 0) {
+                            sttListener.sttEventReceived(new SpeechRecognitionErrorEvent(config.errorMessage));
+                            break;
+                        }
+                        int nSegments = whisper.fullNSegmentsFromState(state);
+                        logger.debug("Available transcription segments {}", nSegments);
+                        if (nSegments == 1) {
+                            tempTranscription = whisper.fullGetSegmentTextFromState(state, 0);
+                            if (config.createWAVRecord) {
+                                createAudioFile(audioSamples, audioSamplesOffset, tempTranscription,
+                                        locale.getLanguage());
+                            }
+                            if (config.singleUtteranceMode) {
+                                logger.debug("single utterance mode, ending transcription");
+                                transcription = tempTranscription;
+                                break;
+                            } else {
+                                // start a new transcription segment
+                                transcription += tempTranscription;
+                                tempTranscription = "";
+                            }
+                        } else if (nSegments == 0 && config.singleUtteranceMode) {
+                            logger.debug("Single utterance mode and no results, ending transcription");
+                            break;
+                        } else if (nSegments > 1) {
+                            // non reachable
+                            logger.error("Whisper should be configured in single segment mode {}", nSegments);
+                            break;
+                        }
+                        // reset state to start with next segment
+                        voiceDetected = false;
+                        silenceSamplesCounter = 0;
+                        audioSamplesOffset = 0;
+                        logger.debug("Partial transcription: {}", tempTranscription);
+                        logger.debug("Transcription: {}", transcription);
+                    }
+                } finally {
+                    if (releaseContext) {
+                        ctx.close();
+                    }
+                }
+                // emit result
+                if (!aborted.get()) {
+                    sttListener.sttEventReceived(new RecognitionStopEvent());
+                    logger.debug("Final transcription: '{}'", transcription);
+                    if (!transcription.isBlank()) {
+                        sttListener.sttEventReceived(new SpeechRecognitionEvent(transcription.trim(), 1));
+                    } else {
+                        sttListener.sttEventReceived(new SpeechRecognitionEvent(" ", 1));
```

Comment:
It should not be SpeechRecognitionErrorEvent in case transcription is blank ?

Reply:
Yes. I did this as a workaround and forgot about it sorry. When emit a SpeechRecognitionErrorEvent the core concat some English text to it and I didn't like to hear that, so I made these workaround so the event was passed to the interpreter and I get the normal "I don't know what to do". I think we should remove that text concatenation from the SpeechRecognitionErrorEvent handler, I'm going to look at it.